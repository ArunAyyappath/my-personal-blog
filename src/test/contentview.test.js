import React from "react";
import ContentDecider from "../component/blogContentViewer";
import { configure, shallow, cleanup, mount, render } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import renderer from "react-test-renderer";
import firebase from "../component/fireStore";
import "firebase/app";
import "firebase/auth";

configure({ adapter: new Adapter() });

beforeAll(async () => {
  const authFire = firebase.auth();
  await authFire.signInWithEmailAndPassword("croft8262@gmail.com", "12345678");
});

afterAll(() => {
  firebase.auth().signOut();
});

afterEach = () => {
  cleanup;
};

describe("<ContentDecider/>", () => {
  it("renders correctly", () => {
    const wrapper = renderer.create(<ContentDecider />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });

  it("ensure static  component have been rendered for the first time", async () => {
    const wrapper = await mount(<ContentDecider />);
    expect(wrapper.find("span").text()).toBe("Your Library Is Empty");
  });
});
