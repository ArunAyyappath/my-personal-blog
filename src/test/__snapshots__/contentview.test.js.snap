// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ContentDecider/> renders correctly 1`] = `
<div
  style={
    Object {
      "width": "100%",
    }
  }
>
  <div
    style={
      Object {
        "float": "left",
        "margin": "50px 40px 10px 60px",
      }
    }
  >
    <div
      className="MuiPaper-root MuiPaper-elevation1 MuiCard-root makeStyles-card-1 MuiPaper-rounded"
    >
      <div
        className="MuiCardHeader-root"
      >
        <div
          className="MuiCardHeader-avatar"
        >
          <div
            aria-label="Recipe"
            className="MuiAvatar-root makeStyles-avatar-5 MuiAvatar-colorDefault"
          >
            Y
          </div>
        </div>
        <div
          className="MuiCardHeader-content"
        >
          <span
            className="MuiTypography-root MuiCardHeader-title MuiTypography-body2 MuiTypography-displayBlock"
          >
            Your Library Is Empty
          </span>
        </div>
      </div>
      <div
        alt="Your Library Is Empty"
        className="MuiCardMedia-root makeStyles-media-2"
        style={
          Object {
            "backgroundImage": "url(\\"blog2N.jpg\\")",
          }
        }
        title="Your Library Is Empty"
      />
      <div
        className="MuiCardContent-root"
      >
        <div
          style={
            Object {
              "height": "20px",
              "overflow": "hidden",
              "width": "313px",
            }
          }
        >
          <p
            className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
          >
            
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    style={
      Object {
        "position": "absolute",
        "right": "1012px",
        "top": "475px",
      }
    }
  >
    <div>
      <div
        className="MuiCircularProgress-root makeStyles-progress-76 MuiCircularProgress-colorSecondary MuiCircularProgress-indeterminate"
        role="progressbar"
        style={
          Object {
            "height": 40,
            "width": 40,
          }
        }
      >
        <svg
          className="MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
            cx={44}
            cy={44}
            fill="none"
            r={20.2}
            strokeWidth={3.6}
            style={Object {}}
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;
